### PR TITLE
Faculty api

### PR DESCRIPTION
**Pull request by zafor-khalid**
1. Basic Faculty related crud 
2. Basic department related crud
3. Removed some linters configuration (prettier and eslinted conflicted in my pc and takes a lot of time to be organized)

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -1,10 +0,0 @@
-module.exports = {
-  root: true,
-  // This tells ESLint to load the config from the package `eslint-config-custom`
-  extends: ['custom'],
-  settings: {
-    next: {
-      rootDir: ['apps/*/'],
-    },
-  },
-};
```
Comment (SayeedAfridi) [inline .eslintrc.js]: why this lines are removed..?

File: .vscode/settings.json [hunk 2]
```diff
@@ -1,14 +1,12 @@
...
-  "editor.codeActionsOnSave": {
-    "source.fixAll.eslint": true
-  },
-  "[typescript]": {
-    "editor.formatOnSave": false
-  },
-  "[typescriptreact]": {
-    "editor.formatOnSave": false
-  },
-  "javascript.preferences.jsxAttributeCompletionStyle": "none",
```
Comment (SayeedAfridi) [inline .vscode/settings.json:11]: same here..!

File: apps/api/src/modules/department/department.controller.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  ): Promise<Department> {
+    return this.service.update(+params.id, data);
+  }
+
+  @Delete(':id')
+  async deleteDepartment(@Param() params) {
+    return this.service.deleteOne(+params.id);
+  }
+
+  @Post('/create')
```
Comment (SayeedAfridi) [inline apps/api/src/modules/department/department.controller.ts]: /create is not the correct rest practice.. it should be empty..

File: apps/api/src/modules/faculty/faculty.controller.ts [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+  ): Promise<Faculty> {
+    return this.service.update(+params.id, data);
+  }
+
+  @Delete(':id')
+  async deleteFaculty(@Param() params) {
+    return this.service.deleteOne(+params.id);
+  }
+
+  @Post('/create')
```
Comment (SayeedAfridi) [inline apps/api/src/modules/faculty/faculty.controller.ts]: same +1

File: apps/api/src/modules/faculty/faculty.module.ts [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+import { PrismaService } from '../persistance/prisma/prisma.service';
+import { FacultyController } from './faculty.controller';
+
+@Module({
+  imports: [],
+  providers: [FacultyService, PrismaService],
+  controllers: [FacultyController],
+  exports: [FacultyService],
+})
+// eslint-disable-next-line prettier/prettier
```
Comment (SayeedAfridi) [inline apps/api/src/modules/faculty/faculty.module.ts:12]: prettier should not complain about this line.. is there any changes in the config files..?

File: packages/database/.env.example [hunk 6]
```diff
@@ -1,3 +1,3 @@
 # database url will be used by prisma
 
-DATABASE_URL="postgresql://username:password@localhost:5432/database?schema=public"
\ No newline at end of file
+DATABASE_URL="postgresql://campus_root:root@66.29.130.137:5432/smart_campus?schema=public"
```
Comment (SayeedAfridi) [inline packages/database/.env.example]: this is not safe.. don't change the example file.. revert it back..

File: packages/database/prisma/schema.prisma [hunk 7]
```diff
@@ -99,4 +99,16 @@ model ModuleRegistry {
...
+
+model Faculty {
+  id              Int                  @id @default(autoincrement())
+  name            String               @unique
+}
+
+model Department {
+  id              Int                  @id @default(autoincrement())
+  name            String               @unique
+  faculty         String
```
Comment (SayeedAfridi) [inline packages/database/prisma/schema.prisma]: faculty should be a relation not String.

**Review by SayeedAfridi — CHANGES_REQUESTED** [review]

File: .eslintrc.js (hunk 1 above)
Comment (zafor-khalid) [reply .eslintrc.js]: it conflicts in my editor, as a result formatting and saving any changes tooks too much time that's why I temporarily removed those

**Review by zafor-khalid — COMMENTED** [review]
all okay

**Pull request merged**